pkg/storegateway/limiter.go

```diff
@@ -39,18 +40,18 @@ type SeriesLimiterFactory func(failedCounter prometheus.Counter) SeriesLimiter
 
 // Limiter is a simple mechanism for checking if something has passed a certain threshold.
 type Limiter struct {
-	limit              uint64
-	reserved           atomic.Uint64
-	errorMessageFormat string
+	limit          uint64
+	reserved       atomic.Uint64
+	limitErrorFunc validation.LimitErrorFunc
```

Comment:
I think we can get rid of the indirection of this func. Can't we just allocate the error when creating the Limiter? Then we know the error message and the exact limit.

Reply:
If we do that then I don't see another use case of having `LimitErrorFunc` at all. In all other occasions we can just have a function which takes a string format and a number and returns `LimitError`.

Reply:
I have addressed your comments and removed `validation.LimitErrorFunc`